### PR TITLE
fix: hide ready message when empty

### LINKED ISSUE #251: workload get includes deliverable information when available
### Description of problem

Workload get includes information about the supply chain process, but does not surface the delivery process information.

The supply chain information can help troubleshoot stuck deployments, and including the delivery information can be part of telling the user the story of the exiting journey thru the the machinery of TAP. 

### Proposed solution

```gherkin
Given I `get` a workload
When the information returns
Then I should see a table with the cluster delivery information
```

### Example

```
Supply chain  
   name:   source-to-url

   RESOURCE           READY   HEALTHY   TIME     OUTPUT  
   Source Provider    True    True      9m ago   ImageRepository/web3
   Image Builder      True    True      8m ago   Deliverable/web3
   Config Provider    True    True      7m ago   Image/web3
   App Config         True    True      6m ago   not found
   Config Writer      True    True      6m ago   not found
 
Delivery 
   name:   delivery-basic

   RESOURCE           READY   HEALTHY   TIME     OUTPUT
   delivery-source    True    True      9m ago   not found
   source-scanner     True    True      8m ago   not found
   source             True    True      7m ago   not found
   testing-pipeline   True    True      6m ago   not found

Messages
   No Messages Found

```

On an **iterate** cluster, the supply chain and delivery information should be available, and both tables can be shown.
On a **build** cluster, the supply chain info would be available, and the delivery table would be omitted
On a **run** cluster, the delivery information would be  available, and the supply table would be omitted

The **messages** section should have messages consolidated from both the workload and deliverable objects.

### Describe alternatives you've considered
Combining supply and delivery steps in a single table. 
* saves a few lines in the workload-get output
* talking to @atmandhol it sounds like it would add complexity to the situations where you would only be able to see either supply or delivery stages, and when outputting yaml.

```
Deployment  
   Supply chain:   source-to-url
   Delivery chain: delivery-basic

   RESOURCE           READY   HEALTH   TIME      OUTPUT
   Source Provider    True    True     9m ago    <value>
   Image Builder      True    True     8m ago    <value>
   Config Provider    True    True     7m ago    <value>
   App Config         True    True     6m ago    <value>
   Config Writer      True    True     6m ago    <value>
   Delivery Source    True    True     6m ago    <value>
   Source Scanner     True    True     6m ago    <value>
   Source             True    True     6m ago    <value>
   Testing Pipeline   True    True     6m ago    <value>

Messages
   No Messages Found


```

### PR DESCRIPTION
**Pull request by odinnordico**
# Pull request

### What this PR does / why we need it

Prevent messages to be printed when the actual condition message is empty

### Which issue(s) this PR fixes

Fixes #251

### CODE REVIEW

File: pkg/printer/workload_deliverable_printer.go [hunk 1]
```diff
@@ -106,15 +106,18 @@ func DeliverableIssuesPrinter(w io.Writer, deliverable *cartov1alpha1.Deliverabl
...
-		readyRow := metav1beta1.TableRow{
-			Cells: []interface{}{
-				fmt.Sprintf("Deliverable [%s]:", readyCondition.Reason),
-				readyCondition.Message,
-			},
+		rows := []metav1beta1.TableRow{}
+		if strings.TrimSpace(readyCondition.Message) != "" {
+			readyRow := metav1beta1.TableRow{
+				Cells: []interface{}{
+					fmt.Sprintf("Deliverable [%s]:", readyCondition.Reason),
```
Comment (shashwathi) [inline pkg/printer/workload_deliverable_printer.go:113]: create const for kind `Deliverable` in pkg cartov1alpha1

File: pkg/printer/workload_status_printer.go [hunk 2]
```diff
@@ -115,15 +115,18 @@ func WorkloadIssuesPrinter(w io.Writer, workload *cartov1alpha1.Workload) error
...
-		readyRow := metav1beta1.TableRow{
-			Cells: []interface{}{
-				fmt.Sprintf("Workload [%s]:", readyCondition.Reason),
-				readyCondition.Message,
-			},
+		rows := []metav1beta1.TableRow{}
+		if strings.TrimSpace(readyCondition.Message) != "" {
+			readyRow := metav1beta1.TableRow{
+				Cells: []interface{}{
+					fmt.Sprintf("Workload [%s]:", readyCondition.Reason),
```
Comment (shashwathi) [inline pkg/printer/workload_status_printer.go:122]: create const or use one if there is for kind Workload in pkg cartov1alpha1

File: pkg/printer/workload_deliverable_printer.go (hunk 1 above)
Comment (odinnordico) [reply pkg/printer/workload_deliverable_printer.go:113]: Addressed in PR #261

File: pkg/printer/workload_status_printer.go (hunk 2 above)
Comment (odinnordico) [reply pkg/printer/workload_status_printer.go:122]: Addressed in PR #261

**Review by shashwathi — APPROVED** [review]

**Pull request merged**